### ISSUE TITLE
Method of switching scheduler node sorting methods in Flight Compute

### ISSUE THREAD

**Issue by wdm78** [alces-software/flight-aws-marketplace#45]
By default, the job-scheduler in Flight compute uses node-spanning queue sort method (i.e. new jobs start on new nodes, and only start filling up part-used nodes when all nodes have at least one job on them). This is optimal for performance, but might end up costing more as we part-use more instances.

This enhancement request is for a simple method to switch to node-packing queue sort method (i.e. new jobs fill up nodes in turn and only spill onto new nodes when part-used nodes are full). Ideally, users would be able to switch sort methods from the command-line.

**Comment by sierra-tango-echo** [comment]
this is interesting, we may have the default the wrong way around - on scaling clusters at least? @mjtko whats your opinion?

**Comment by mjtko** [comment]
Yes, seems to be the wrong way round for AWS autoscaling clusters.

I think this boils down to two enhancements:
1. Add a tool to switch the allocation method (`alces configure scheduler`?)
2. Add a configuration flag that allows us to affect how this is configured at boot:

```
cluster:
  […]
  scheduler:
    allocation: <packing|spanning|autodetect>
```

Perhaps we do some kind of auto-detection depending on how the environment is configured.  We could implement this as a heuristic that chooses `packing` if the cluster is in scaling mode or uses `spanning` otherwise.

(This is trying to remain scheduler-agnostic, and makes the assumption that all schedulers have this kind of configuration option.)

**Comment by mjtko** [comment]
Fixed in alces-software/clusterware@1f753e9

Can now do this by default by supplying appropriate config.yml keys (`cluster`->`scheduler`->`allocation`), or manually enable via `alces configure scheduler allocation` command. Possible values are `packing` and `spanning`.

**Comment by vlj91** [comment]
This is included in `2016.3`
